shared/src/libs/helpers/debounce/debounce.helper.ts

```diff
@@ -0,0 +1,22 @@
+import libraryDebounce from 'debounce';
+import { type Dispatch, type SetStateAction } from 'react';
+
+const DEBOUNCE_DELAY = 300;
+
+let debounced: ReturnType<typeof libraryDebounce> | undefined;
+
+const debounce = (
+  setSearchQuery: Dispatch<SetStateAction<string>>,
+): ReturnType<typeof libraryDebounce> => {
+  if (debounced) {
+    debounced.clear();
+  }
```

Comment:
this won't work, we will have one state per all denounced functions, we need to call this inside component
```suggestion
```

Reply:
Sorry could you clarify this for me a little bit? Now It works like this: debounce function debounces setting the state of searchQuery. And filtering starts only when searchQuery is set. It also clears timer calling clear. As I understood I need to change it by adding state to the input-search and use filtering logic there, and also I have to debounce filtering logic but not the state setting?

Reply:
we can't store debounced function link inside helper file, because we may have different denounced functions at the same time. So the main point here is call clear on returned function where it's used, not inside helper. You can use component state if it's needed

Reply:
Thank you, I got it now

Reply:
You can follow the implementation from pr on web https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/77/files#diff-cc19d0b1115b807f2d2964a4a1f30b1246a5caec87ba80cc11de61076fd7f203

Reply:
I changed the debounce function according to your comments and the PR that you mentioned. It works, but I am not sure that I use it in the best way, I will work on it tomorrow.